platform/mbed_rtc_time.cpp

```diff
@@ -94,24 +112,45 @@ time_t time(time_t *timer)
     if (_rtc_read != NULL) {
         t = _rtc_read();
     }
+    
+    tv->tv_sec = t;  
+        
+    _mutex->unlock();
 
-    if (timer != NULL) {
-        *timer = t;
+    // timezone struct is deprecated
+    if (tz) {
+        MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM,  MBED_ERROR_CODE_INVALID_ARGUMENT), 
+            "timezone parameter not null, argument has been deprecated and will not be modified");
+        return 1;
     }
-    _mutex->unlock();
-    return t;
+
+    return 0;
 }
 
-void set_time(time_t t)
+#if defined (__ICCARM__)
+time_t __time32(time_t *timer)
+#else
+time_t time(time_t *timer)
+#endif
 {
-    _mutex->lock();
-    if (_rtc_init != NULL) {
-        _rtc_init();
+    struct timeval tv;
+    int32_t err = gettimeofday(&tv, NULL);
+ 
+    if (err) {
+        return err;
```

Comment:
If `gettimeofday` did return an error, `time()` should return+output `(time_t)-1`.

As it stands, in error cases, `tv_sec` is -1, so you can remove this test and let it just always return+output `tv_sec`.